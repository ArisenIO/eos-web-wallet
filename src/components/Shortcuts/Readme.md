A list of shortcuts example:

```js


<div className="shortcuts-example">
  <Shortcuts renderItem={Shortcut2} />
</div>
```
