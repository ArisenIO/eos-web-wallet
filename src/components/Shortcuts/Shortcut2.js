import * as React from "react";
import { NavLink } from "react-router-dom";

const Shortcut2 = ({
  className = 'col-link px-4 py-3',
  iconClass,
  text,
  ...props }) => (
  <li>
    <a
      href='#'
      className={className}
      exact
      {...props}
    >
      <span className={iconClass} />
      {text}
    </a>
  </li>
);

export default Shortcut2;