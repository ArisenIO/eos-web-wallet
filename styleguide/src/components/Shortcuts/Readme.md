A list of shortcuts example:

```js
<div className="shortcuts-example">
  <Shortcuts />
</div>
```
