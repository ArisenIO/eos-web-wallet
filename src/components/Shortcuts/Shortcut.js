import * as React from "react";
import { NavLink } from "react-router-dom";
import Shortcuts from "components/Shortcuts";
import css from "./styles.module.scss";

const Shortcut = (item, _key) => {
  if (Number.isInteger(_key)) {
    return (
      <li key={_key}>
        <Shortcuts data={item} />
      </li>
    );
  } else {
    const {
      iconClass,
      text,
      key,
      ...props } = item;

    return (
      <li key={key}>
        <NavLink
          exact
          {...props}
        >
          <span className={iconClass} />
          {text}
        </NavLink>
      </li>
    );
  }
}

export default Shortcut;
