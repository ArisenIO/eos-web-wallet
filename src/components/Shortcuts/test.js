import * as React from "react";
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import Shortcuts from './';

describe('<Shortcuts />', () => {
  it('renders without crashing', () => {
    const staticContext = {};
    const tree = renderer.create(
      <StaticRouter location="/" context={staticContext}>
        <Shortcuts />
      </StaticRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})


