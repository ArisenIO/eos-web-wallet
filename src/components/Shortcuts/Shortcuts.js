import * as React from "react";
import List from "components/List";
import Shortcut from "./Shortcut";
import cx from "classnames";
import css from "./styles.module.scss";

const activeClassName = 'active';
const links = [
  [
    {
      to: '/',
      text: 'Transfer',
      iconClass: 'icon-transfer u-mr1',
      activeClassName,
    },
    {
      to: '/transactions',
      text: 'Transaction History',
      iconClass: 'icon-history u-mr1',
      activeClassName,
    },
    {
      to: '/permissions',
      text: 'Permissions',
      iconClass: 'icon-permissions u-mr1',
      activeClassName,
    },
    {
      to: '/logout',
      text: 'Logout',
      iconClass: 'icon-logout u-mr1',
      activeClassName,
    },
  ],
  {
    to: '/users',
    text: 'Users',
    activeClassName,
  },
  {
    to: '/about',
    text: 'About',
    activeClassName,
  },
  {
    to: '/faq',
    text: 'FAQ',
    activeClassName,
  },
];

const Shortcuts = ({
  className,
  data,
  renderItem = Shortcut }) => (
  <List
    className="menulist"
    data={data || links}
    renderItem={renderItem}
  />
);

export default Shortcuts;
