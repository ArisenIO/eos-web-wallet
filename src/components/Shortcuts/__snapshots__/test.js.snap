// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shortcuts /> renders without crashing 1`] = `
<ul
  className="-links"
>
  <li>
    <a
      aria-current="true"
      className="col-link px-4 py-3 active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <span
        className="icon-eos_icons_transfer mr-2"
      />
      Transfer
    </a>
  </li>
  <li>
    <a
      aria-current={false}
      className="col-link px-4 py-3"
      href="/transactions"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="icon-eos_icons_history mr-2"
      />
      Transaction History
    </a>
  </li>
  <li>
    <a
      aria-current={false}
      className="col-link px-4 py-3"
      href="/permissions"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="icon-eos_icons_permissions mr-2"
      />
      Permissions
    </a>
  </li>
  <li>
    <a
      aria-current={false}
      className="col-link px-4 py-3"
      href="/logout"
      onClick={[Function]}
      style={undefined}
    >
      <span
        className="icon-eos_icons_logout mr-2"
      />
      Logout
    </a>
  </li>
</ul>
`;
